src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -1,654 +1,650 @@
+                double damage = getRangedAttackDamage();
+                final double chance = HIT_CHANCE_DIVIDER / (worker.getCitizenData().getLevel() + 1);
+
+                arrow.shoot(xVector, yVector + distance * RANGED_AIM_SLIGHTLY_HIGHER_MULTIPLIER, zVector, (float) 1.6D, (float) chance);
+
+                if (worker.getHealth() <= DOUBLE_DAMAGE_THRESHOLD)
+                {
+                    damage *= 2;
+                }
+
+                arrow.setDamage(damage);
+                final double xDiff = target.posX - worker.posX;
+                final double zDiff = target.posZ - worker.posZ;
+                final double goToX = xDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
+                final double goToZ = zDiff > 0 ? MOVE_MINIMAL : -MOVE_MINIMAL;
+
+                worker.move(MoverType.SELF, goToX, 0, goToZ);
+                worker.playSound(SoundEvents.ENTITY_SKELETON_SHOOT, (float) BASIC_VOLUME, (float) getRandomPitch());
+                worker.world.spawnEntity(arrow);
+
+                target.setRevengeTarget(worker);
+
+                worker.damageItemInHand(EnumHand.MAIN_HAND, 1);
+            }
         }
-        return pos;
+        return GUARD_ATTACK_RANGED;
     }
 
     /**
-     * Searches a random direction.
-     *
-     * @param random a random object.
-     * @return a tuple of two directions.
+     * Gets the reload time for a Range guard attack.
+     * @return the reload time
      */
-    private Tuple<EnumFacing, EnumFacing> getRandomDirectionTuple(final Random random)
+    protected int getAttackDelay()
     {
-        if (lastDirection != null && random.nextInt(IN_TEN) < LAST_DIRECTION_CHANCE)
+        if (worker.getCitizenData() != null)
         {
-            return lastDirection;
+            return RANGED_ATTACK_DELAY_BASE / (worker.getCitizenData().getLevel() + 1);
         }
-
-        return new Tuple<>(EnumFacing.random(random), EnumFacing.random(random));
+        return RANGED_ATTACK_DELAY_BASE;
     }
 
     /**
-     * Checks if the the worker is too far from his patrol/guard/follow target.
-     *
-     * @param target an attack target.
-     * @param range  the range allowed to be from the patrol/guard/follow target.
-     * @return true if too far.
+     * Gets the aim height for ranged guards.
+     * @return the aim height.
+     * Suppression because the method already explains the value.
      */
-    public boolean shouldReturnToTarget(final BlockPos target, final double range)
+    @SuppressWarnings("squid:S3400")
+    protected double getAimHeight()
     {
-        final AbstractBuilding building = getOwnBuilding();
-        if (currentPathTarget == null)
-        {
-            getNextPatrollingTarget((AbstractBuildingGuards) building);
-        }
-
-        return building instanceof AbstractBuildingGuards
-                 && BlockPosUtil.getDistance2D(target, currentPathTarget) > ((AbstractBuildingGuards) building).getPatrolDistance() + range;
+        return 3.0D;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 3.0D to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)